app/services/s3_zip_downloader.rb

```diff
@@ -14,6 +14,7 @@ def generate!
   private
 
   def build_zip_file(documents, bundle)
+    documents.includes(:document_blob, :converted_preview_document_attachment) # Eager load associations
```

Comment:
It might be better to do the eager loading when `documents` are first queried on line 10 - eg:

```
build_zip_file @claim.documents.includes(:document_blob, :converted_preview_document_attachment), bundle
```

I think your approach will result in two database calls, whereas this will do it all in a single call.